### PR TITLE
Optimization + Fixes

### PR DESCRIPTION
**Pull request by tgiphil**
Includes a move optimization when SSA is no possible; such as for exception handling.

### CODE REVIEW

File: Source/Mosa.Compiler.Framework/Stages/SparseConditionalConstantPropagationStage.cs [hunk 1]
```diff
@@ -49,6 +51,9 @@ protected override void Run()
 		if (HasProtectedRegions)
 			return;
 
+		Steps = 0;
+		MethodCompiler.CreateTranformInstructionTrace(this, Steps++);
```
Comment (charsleysa) [inline Source/Mosa.Compiler.Framework/Stages/SparseConditionalConstantPropagationStage.cs:55]: Is this intended to be 0 or 1 for the input parameter? Currently it'll input 0 as the increment happens afterwards.

**Review by charsleysa — APPROVED** [review]
LGTM

File: Source/Mosa.Compiler.Framework/Stages/SparseConditionalConstantPropagationStage.cs (hunk 1 above)
Comment (tgiphil) [reply Source/Mosa.Compiler.Framework/Stages/SparseConditionalConstantPropagationStage.cs:55]: Should start at 0

**Review by charsleysa — APPROVED** [review]
LGTM

**Pull request merged**